libs/vkd3d/command.c

```diff
@@ -3442,21 +3442,19 @@ static void d3d12_command_list_end_current_render_pass(struct d3d12_command_list
                 list->so_counter_buffers, list->so_counter_buffer_offsets));
     }
 
-    if (list->current_render_pass)
+    if (list->rendering_info.state_flags & VKD3D_RENDERING_ACTIVE)
     {
-        VkSubpassEndInfoKHR subpass_end_info;
-        subpass_end_info.sType = VK_STRUCTURE_TYPE_SUBPASS_END_INFO_KHR;
-        subpass_end_info.pNext = NULL;
+        list->rendering_info.state_flags &= ~VKD3D_RENDERING_ACTIVE;
 
-        VK_CALL(vkCmdEndRenderPass2KHR(list->vk_command_buffer, &subpass_end_info));
+        VK_CALL(vkCmdEndRenderingKHR(list->vk_command_buffer));
     }
 
     /* Don't emit barriers for temporary suspension of the render pass */
-    if (!suspend && (list->current_render_pass || list->render_pass_suspended))
+    if (!suspend && (list->rendering_info.state_flags & (VKD3D_RENDERING_ACTIVE | VKD3D_RENDERING_SUSPENDED)))
         d3d12_command_list_emit_render_pass_transition(list, VKD3D_RENDER_PASS_TRANSITION_MODE_END);
 
-    list->render_pass_suspended = suspend && (list->current_render_pass || list->render_pass_suspended);
-    list->current_render_pass = VK_NULL_HANDLE;
+    if (suspend && (list->rendering_info.state_flags & (VKD3D_RENDERING_ACTIVE)))
```

Comment:
This branch should never hit I think?

Reply:
Maybe it's supposed to check CURRENT flag?

Reply:
See above.